### PR TITLE
User Permissions and admin

### PR DESCRIPTION
**Pull request by loppear**
Adds permissions and roles-as-list-of-permissions, and a User Admin.

The role admin needs some love still, and I'm not creating the default roles yet, but figured it's ready for some eyes. (Can be installed in 100mhl-platform or app-boilerplate and _fingers crossed_ just works.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,8 +6,8 @@
   "scripts": {
     "preversion": "npm test",
     "postversion": "npm run build-docs && git push && git push --tags",
-    "test": "NODE_ENV=test mocha --compilers js:babel-core/register -R spec test/*.js src/modules/**/test/*.js",
-    "compile": "babel src --out-dir lib",
+    "test": "NODE_ENV=test mocha --compilers js:babel-register -R spec test/*.js src/modules/**/test/*.js",
+    "compile": "babel src --out-dir lib; cp -r src/modules/users-admin/templates lib/modules/users-admin/",
```
Comment (alexjreich) [inline package.json]: We really need a better way to handle copying files around...

File: src/models/roleModel.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import {BaseModel} from 'nxus-storage'
+
+export default BaseModel.extend({
```
Comment (alexjreich) [inline src/models/roleModel.js:3]: Didn't we have a method of using the native `extend` for these base model classes?

File: src/modules/users-permissions/PermissionManager.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    // construct user permissions list
+    this._permissions = new Set()
+    this._user.roles.forEach((role) => {
+      for (let p of role.permissions) {
+        this._permissions.add(p)
+      }
+    })
+  }
+
+  allow(permissionName, obj = null) {
```
Comment (alexjreich) [inline src/modules/users-permissions/PermissionManager.js]: This has a different intent than the `UsersPermissions.allow()` correct?  One is a register function, the other is a check?  If so, I would make consistent or rename to clarify the different functionality.

**Review by alexjreich — COMMENTED** [review]
A couple of notes, overall looks like a good base implementation.  Have you tested with 100MHL yet?

File: src/models/roleModel.js (hunk 2 above)
Comment (loppear) [reply src/models/roleModel.js:3]: Not that I've seen, I'll add an issue to investigate though.

File: src/modules/users-permissions/PermissionManager.js (hunk 3 above)
Comment (loppear) [reply src/modules/users-permissions/PermissionManager.js]: Ah yep. The proposal in #20 was to name this `allowed` (`req.user.permissions.allowed()` now, does that still sound too close? `allows` read better but is very close :) (As discussed there, not a good idea for this runtime call to support the same permissions-registering intent.)

**Comment by loppear** [comment]
Tested on 100MHL in terms of the default user admin permissions and middleware not getting in the way - going in now and testing actually adding permissions to application-specific routes.

**Pull request merged**